### PR TITLE
Live signals

### PR DESCRIPTION
**Pull request by moustaphaalaa**

### CODE REVIEW

File: packages/clarity-js/types/data.d.ts [hunk 1]
```diff
@@ -401,3 +402,9 @@ export interface UploadData {
     attempts: number;
     status: number;
 }
+
+export interface ClaritySignal {
+    type: string
+    timestamp?: string
```
Comment (OmarBazaraa) [inline packages/clarity-js/types/data.d.ts:408]: This is sent as a Unix time milliseconds. A `number` will suit here more.

**Review by AbdelrhmanMagdy — APPROVED** [review]

**Comment by moustaphaalaa** [comment]
@microsoft-github-policy-service agree company="Microsoft"

File: packages/clarity-js/src/core/signal.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+function parseSignals(signalsString: string): ClaritySignal[] {
+    const signalsJson: ClaritySignal[] = JSON.parse(signalsString)
+    return signalsJson;
+}
+
+export function signalEvent(signalsString: string) {
+    try {
+        const signals = parseSignals(signalsString);
+        signals.forEach(signal => {
+            signalCallback(signal)
```
Comment (swaathee) [inline packages/clarity-js/src/core/signal.ts:13]: check whether signalcallback exists

**Review by swaathee — CHANGES_REQUESTED** [review]

**Comment by moustaphaalaa** [comment]
@microsoft-github-policy-service agree company="Microsoft"

File: packages/clarity-js/src/data/signal.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+function parseSignals(signalsPayload: string): ClaritySignal[] {
+  try{
+    const parsedSignals: ClaritySignal[] = JSON.parse(signalsPayload);
+    return parsedSignals;
+  }catch{
+    return []
+  }
+}
+
+export function triggerSignalsEvent(signalsPayload: string) {
```
Comment (swaathee) [inline packages/clarity-js/src/data/signal.ts:18]: nit: prefer signalEvent to triggerSignalsEvent

**Review by swaathee — APPROVED** [review]

**Pull request merged**